

services:
  surya_app:
    image: xiaoyao9184/surya:0.14.0
    container_name: surya_ocr_app
    environment:
      - TORCH_DEVICE=cpu
      - HF_HUB_OFFLINE=true
      - DETECTOR_MODEL_CHECKPOINT=/root/.cache/datalab/models/text_detection/2025_05_07
      - LAYOUT_MODEL_CHECKPOINT=/root/.cache/datalab/models/layout/2025_02_18
      - RECOGNITION_MODEL_CHECKPOINT=/root/.cache/datalab/models/text_recognition/2025_05_13
      - TABLE_REC_MODEL_CHECKPOINT=/root/.cache/datalab/models/table_recognition/2025_02_18
      - OCR_ERROR_MODEL_CHECKPOINT=/root/.cache/datalab/models/ocr_error_detection/2025_02_18
      - TEXIFY_MODEL_CHECKPOINT=/root/.cache/datalab/models/texify/2025_02_18
      # - FONT_DIR=/fonts # not work
    ports:
      - "8501:8501"
    volumes:
      - ./../../cache:/root/.cache
      # - ./../../static/fonts:/fonts # not work
      # - ./../../static/fonts:/opt/conda/lib/python3.11/site-packages/static/fonts # image already included
